python/ray/serve/context.py

```diff
@@ -205,3 +207,23 @@ def _set_request_context(
             or current_request_context.multiplexed_model_id,
         )
     )
+
+
+# A map from current request ID to a set of ReplicaResults corresponding
+# to the requests the replica has sent during the current request
+_requests_pending_assignment: Dict[str, Dict[str, asyncio.Task]] = defaultdict(dict)
+
+
+def _get_requests_pending_assignment(parent_request_id: str) -> Dict:
+    if parent_request_id in _requests_pending_assignment:
+        return _requests_pending_assignment[parent_request_id]
+
+
+def _add_request_pending_assignment(parent_request_id: str, response_id: str, task):
+    if parent_request_id:
```

Comment:
This if statement will probably always be true? (if we switch to use `internal_request_id`) :)